st2client/st2client/commands/action_alias.py

```diff
@@ -40,3 +54,106 @@ class ActionAliasGetCommand(resource.ContentPackResourceGetCommand):
     display_attributes = ['all']
     attribute_display_order = ['id', 'ref', 'pack', 'name', 'description',
                                'enabled', 'action_ref', 'formats']
+
+
+class ActionAliasMatchCommand(resource.ResourceCommand):
+    display_attributes = ['id', 'name', 'description']
+
+    def __init__(self, resource, *args, **kwargs):
+        super(ActionAliasMatchCommand, self).__init__(
+            resource, 'match',
+            'Get the list of %s that match the command text.' %
+            resource.get_plural_display_name().lower(),
+            *args, **kwargs)
+
+        self.parser.add_argument('match_text',
+                                 metavar='command',
+                                 help=help)
+
+        self.parser.add_argument('-a', '--attr', nargs='+',
+                                 default=self.display_attributes,
+                                 help=('List of attributes to include in the '
+                                       'output. "all" will return all '
+                                       'attributes.'))
+        self.parser.add_argument('-w', '--width', nargs='+', type=int,
+                                 default=None,
+                                 help=('Set the width of columns in output.'))
+
+    @resource.add_auth_token_to_kwargs_from_cli
+    def run(self, args, **kwargs):
+        aliases = self.manager.get_all(**kwargs)
+        matches = match_command_to_alias(args.match_text, aliases)
+        return [match[0] for match in matches]  # show only alias objects
+
+    def run_and_print(self, args, **kwargs):
+        instances = self.run(args, **kwargs)
+        self.print_output(instances, table.MultiColumnTable,
+                          attributes=args.attr, widths=args.width,
+                          json=args.json, yaml=args.yaml)
+
+
+class ActionAliasExecuteCommand(ActionRunCommandMixin,
+                                resource.ResourceCommand):
+    def __init__(self, resource, *args, **kwargs):
```

Comment:
Nice, I like the idea of having a CLI command for those two operations - should make testing some scenarios easier :+1: